cpp/src/groupby/sort/group_scan_util.cuh

```diff
@@ -192,43 +190,19 @@ struct group_scan_functor<K,
   {
     if (values.is_empty()) { return cudf::empty_like(values); }
 
-    // When finding MIN, we need to consider nulls as larger than non-null elements.
-    // Thing is opposite when finding MAX.
-    auto const null_precedence  = (K == aggregation::MIN) ? null_order::AFTER : null_order::BEFORE;
-    auto const flattened_values = structs::detail::flatten_nested_columns(
-      table_view{{values}}, {}, std::vector<null_order>{null_precedence});
-    auto const d_flattened_values_ptr = table_device_view::create(flattened_values, stream);
-    auto const flattened_null_precedences =
-      (K == aggregation::MIN)
-        ? cudf::detail::make_device_uvector_async(flattened_values.null_orders(), stream)
-        : rmm::device_uvector<null_order>(0, stream);
+    auto constexpr is_min_op = K == aggregation::MIN;
+    auto const binop_generator =
+      cudf::reduction::detail::comparison_binop_generator(values, is_min_op, stream);
```

Comment:
It would be a little cleaner to pass the aggregation to the `comparison_binop_generator` and have it do the `is_min_op` logic internally.  That way the user doesn't have to type it out every time.

Reply:
Done.